### PR TITLE
doc: warn that incoming conns are unlikely when not using default ports

### LINKED ISSUE #5150: Should a warning be thrown when using -port that nodes are very unlikely to get any incoming connections?
It's surprising that using a non-standard listening port will cause peers to actively avoid connecting to you, is it sensible to add this to the help message and the warnings on startup if people have changed the default? I can't find this behavior mentioned anywhere except for the comments in net.cpp. 

```
// do not allow non-default ports, unless after 50 invalid addresses selected already
if (addr.GetPort() != Params().GetDefaultPort() && nTries < 50)
continue;
```

https://github.com/bitcoin/bitcoin/blob/master/src/net.cpp#L1369

### PR DESCRIPTION
**Pull request by adamjonas**
Closes #5150.

This was mostly copied from #5285 by sulks, who has since quit GitHub.

The issue has remained open for 6 years, but the extra explanation still seems useful.

### CODE REVIEW

**Comment by laanwj** [comment]
ACK /b2bbc089888015e86ce3cbe145fd5a61695205ff

File: src/init.cpp [hunk 1]
```diff
@@ -450,7 +450,7 @@ void SetupServerArgs(NodeContext& node)
...
     argsman.AddArg("-peerblockfilters", strprintf("Serve compact block filters to peers per BIP 157 (default: %u)", DEFAULT_PEERBLOCKFILTERS), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-permitbaremultisig", strprintf("Relay non-P2SH multisig (default: %u)", DEFAULT_PERMIT_BAREMULTISIG), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-port=<port>", strprintf("Listen for connections on <port> (default: %u, testnet: %u signet: %u, regtest: %u)", defaultChainParams->GetDefaultPort(), testnetChainParams->GetDefaultPort(), signetChainParams->GetDefaultPort(), regtestChainParams->GetDefaultPort()), ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-port=<port>", strprintf("Listen for connections on <port>. Nodes not using the default ports (default: %u, testnet: %u, signet: %u, regtest: %u) will be unlikely to get incoming connections.", defaultChainParams->GetDefaultPort(), testnetChainParams->GetDefaultPort(), signetChainParams->GetDefaultPort(), regtestChainParams->GetDefaultPort()), ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::CONNECTION);
```
Comment (jonatack) [inline src/init.cpp:453]: thanks for fixing up the grammar while here

tried to make this part more easy/clear, not sure it's better
```suggestion
    argsman.AddArg("-port=<port>", strprintf("Listen for connections on <port>. Nodes not using the default ports (default: %u, testnet: %u, signet: %u, regtest: %u) will likely not see incoming connections.", defaultChainParams->GetDefaultPort(), testnetChainParams->GetDefaultPort(), signetChainParams->GetDefaultPort(), regtestChainParams->GetDefaultPort()), ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::CONNECTION);
```

**Review by jonatack — COMMENTED** [review]
utACK b2bbc089888015e86ce3cbe145fd5a61695205ff

**Comment by MarcoFalke** [comment]
ACK b2bbc089888015e86ce3cbe145fd5a61695205ff

Nice find

File: src/init.cpp (hunk 1 above)
Comment (practicalswift) [inline src/init.cpp:453]: Nit: "are" instead of "will be"?

**Comment by adamjonas** [comment]
[Updated](https://github.com/bitcoin/bitcoin/compare/b2bbc089888015e86ce3cbe145fd5a61695205ff..010eed3ce03cf4fc622a48f40fc4d589383f7a44) to go with @practicalswift's readability recommendation.

Thanks for the quick review and suggestions.

**Pull request merged**